bson/marshalvalue_cases_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
```

Comment:
Should this file be "bson/marshalvalue_cases_test.go" or "bson/marshal_value_cases_test.go" to (sort of) name it after the "marshalValueTestCase" struct?